### PR TITLE
deps: bump typescript to 4.8, fix Dictionary typings

### PR DESCRIPTION
**Pull request by mayorovad**
**Summary**
After updating to typescript 4.8, when running `generate-types` script in core package there is error TS2345 in src/util/Dictionary.ts, added changes to fix it. 

**Description for the changelog**
- Bumped Typescript to 4.8.3 and typedoc to 0.23.15
- Moved types IdentityObject and IdentityFunction to types.ts
- Renamed const FIELD_NAME to IDENTITY_FIELD_NAME and moved it to util/Constants.ts
- Set Cell and CellOverlay to implement IdentityObject (found only them using in Dictionary as key)

### CODE REVIEW

File: packages/core/src/util/Dictionary.ts [hunk 1]
```diff
@@ -16,6 +16,7 @@ See the License for the specific language governing permissions and
 limitations under the License.
 */
 
+import { IdentityFunction, IdentityObject } from 'src/types';
```
Comment (junsikshim) [inline packages/core/src/util/Dictionary.ts]: How about using relative path something like '../types' instead?

File: packages/core/src/view/cell/CellOverlay.ts [hunk 2]
```diff
@@ -21,6 +21,7 @@ import Rectangle from '../geometry/Rectangle';
 import EventSource from '../event/EventSource';
 import ImageBox from '../image/ImageBox';
 import CellState from './CellState';
+import ObjectIdentity from 'src/util/ObjectIdentity';
```
Comment (junsikshim) [inline packages/core/src/view/cell/CellOverlay.ts]: Also here.

**Review by junsikshim — COMMENTED** [review]
Good work!
I've got to admit I haven't put much effort in ObjectIdentity.
It's great to see more people contributing. :)

**Comment by mayorovad** [comment]
@junsikshim thanks for review! Sorry, didn't noticed absolute paths when pushing (maybe add `eslint` to catch thinks like that?)

**Review by tbouffard — APPROVED** [review]
Sorry for the late review, I wanted to first work on #128 as we are bumping TypeScript here.
I reviewed the generated types with the newest TypeScript version and I didn't notice major changes because of the new version.

The only thing I noticed is that before, `SelectionCellsHandler.d.ts` included
```typescript
getHandler(cell: Cell): EdgeHandler | VertexHandler | null;
```
Now it is (notice the addition of `NonNullable`)
```typescript
getHandler(cell: Cell): NonNullable<EdgeHandler | VertexHandler> | null;
```

As `NonNullable` has been available since [TS 2.8](https://www.typescriptlang.org/docs/handbook/release-notes/typescript-2-8.html#predefined-conditional-types), this won't change the minimum TS version required by `maxGraph`.

**Pull request merged**